pygpu/tools.py

```diff
@@ -168,58 +168,6 @@ def check_args(args, collapse=False, broadcast=False):
     return n, nd, dims, tuple(strs), tuple(offsets)
 
 
-class Counter(dict):
-    'Mapping where default values are zero'
-    def __missing__(self, key):
-        return 0
-
-
-def lfu_cache(maxsize=20):
```

Comment:
I don't know.  Because that is what was working.  Also, that cache is currently only used the python code for reduction which is hopefully going away soon.

I don't want to put much effort in it.